static/test/javascripts/spec/common/utils/timeout.spec.js

```diff
@@ -0,0 +1,58 @@
+define([
+    'Promise',
+    'common/utils/timeout'
+], function (
+    Promise,
+    timeout
+) {
+    function wait (time) {
+        return new Promise(function (resolve) {
+            setTimeout(resolve, time);
+        });
+    }
+
+    describe('Timeout utility', function () {
+        it('resolves with the promised value if the timeout doesn\'t fire', function (done) {
+            timeout(10, Promise.resolve(12))
+            .then(function (result) {
+                expect(result).toBe(12);
+                return wait(10);
```

Comment:
internally the `timeout` function is still firing the `setTimeout` after 10ms, so I prefer to wait for that to happen before finishing the test